src/parser/tests/class_parser_tests.rs

```diff
@@ -340,3 +340,176 @@ fn fb_method_with_return_type_can_be_parsed() {
+            VAR
+                x : DINT;
+                y : DINT := 3;
+            END_VAR
+            VAR_TEMP
+                xTmp : DINT;
+                yTmp : DINT := 3;
+            END_VAR
+            VAR_INPUT
+                xIn : DINT;
+                yIn : DINT := 3;
+            END_VAR
+            VAR_OUTPUT
+                xOut : DINT;
+                yOut : DINT := 3;
+            END_VAR
+            VAR_IN_OUT
+                xIO : DINT;
+                yIO : DINT;
+            END_VAR
+        END_METHOD
+    END_FUNCTION_BLOCK
+
+    PROGRAM prg
+        METHOD mthd
+            VAR
+                x : DINT;
+                y : DINT := 3;
+            END_VAR
+            VAR_TEMP
+                xTmp : DINT;
+                yTmp : DINT := 3;
+            END_VAR
+            VAR_INPUT
+                xIn : DINT;
+                yIn : DINT := 3;
+            END_VAR
+            VAR_OUTPUT
+                xOut : DINT;
+                yOut : DINT := 3;
+            END_VAR
+            VAR_IN_OUT
+                xIO : DINT;
+                yIO : DINT;
+            END_VAR
+        END_METHOD
+    END_PROGRAM
+    ";
+
+    let (unit, _) = parse(src);
+    let fb_mthd = &unit.units[1];
+    assert_eq!(fb_mthd.name, "fb.mthd".to_string());
+    assert_eq!(fb_mthd.pou_type, PouType::Method { owner_class: "fb".into() });
+
+    let prg_mthd = &unit.units[3];
+    assert_eq!(prg_mthd.name, "prg.mthd".to_string());
+    assert_eq!(prg_mthd.pou_type, PouType::Method { owner_class: "prg".into() });
+
+    // we expect one of each of these `VariableBlockType` to be parsed
+    let expected_var_blocks = vec![
+        VariableBlockType::Local,
+        VariableBlockType::Temp,
+        VariableBlockType::Input(ArgumentProperty::ByVal),
+        VariableBlockType::Output,
+        VariableBlockType::InOut,
+    ];
+    let actual =
+        &fb_mthd.variable_blocks.iter().map(|it| it.variable_block_type).collect::<Vec<VariableBlockType>>();
+    assert_eq!(&expected_var_blocks, actual);
+    let actual =
+        &prg_mthd.variable_blocks.iter().map(|it| it.variable_block_type).collect::<Vec<VariableBlockType>>();
+    assert_eq!(&expected_var_blocks, actual);
+
+    // we expect to have parsed 10 variables and 4 of them (all `y`s apart from the in-out) to have initializer
+    let variables = &fb_mthd.variable_blocks.iter().fold(vec![], |mut acc, block| {
+        acc.extend(block.variables.iter());
+        acc
+    });
+    assert_eq!(variables.len(), 10);
+    let with_initializer = variables.iter().filter(|it| it.initializer.is_some());
+    assert_eq!(with_initializer.count(), 4);
+
+    let variables = &prg_mthd.variable_blocks.iter().fold(vec![], |mut acc, block| {
+        acc.extend(block.variables.iter());
+        acc
+    });
+    assert_eq!(variables.len(), 10);
+    let with_initializer = variables.iter().filter(|it| it.initializer.is_some());
+    assert_eq!(with_initializer.count(), 4);
```

Comment:
Hats off for unit testing this way, my lazy ass would've slapped a `insta::assert_snapshot` there 😅